/usr/bin/ld: cannot find -lval-NOTFOUND

When building the snapshot from 2020-07-06 locally on Linux, I get this error:
```
Scanning dependencies of target encrypt
[ 64%] Building C object src/examples/CMakeFiles/encrypt.dir/encrypt.c.o
[ 65%] Linking C executable encrypt
/usr/bin/ld: cannot find -lval-NOTFOUND
collect2: error: ld returned 1 exit status
make[2]: *** [src/examples/CMakeFiles/encrypt.dir/build.make:85: src/examples/encrypt] Error 1
make[1]: *** [CMakeFiles/Makefile2:101: src/examples/CMakeFiles/encrypt.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
```

cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/usr/local/botan-rnp-shared -DBUILD_SHARED_LIBS=on -DBUILD_TESTING=off ../rnp

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,7 +28,8 @@
 # 3.10+ for gtest_discover_tests (parallel rnp_tests)
 # 3.12+ for NAMELINK_COMPONENT (for better RPM packaging)
 # 3.12+ for Python3 find module
-cmake_minimum_required(VERSION 3.12)
+# 3.14+ for object library link dependency propagation
+cmake_minimum_required(VERSION 3.14)
 
 # contact email, other info
 include(cmake/info.cmake)

